src/transformers/integrations.py

```diff
@@ -1137,6 +1162,23 @@ def on_train_end(self, args, state, control, **kwargs):
     def __del__(self):
         self._stop_run_if_exists()
 
+    def on_save(self, args, state, control, **kwargs):
+        if self._should_upload_checkpoint:
+            checkpoint_path = os.path.join(args.output_dir, f"checkpoint-{state.global_step}")
+            self._log_model_checkpoint(checkpoint_path)
+
+    def on_evaluate(self, args, state, control, metrics=None, **kwargs):
+        if self._log_checkpoints == 'best':
+            best_metric_name = args.metric_for_best_model
+            if not best_metric_name.startswith("eval_"):
+                best_metric_name = f"eval_{best_metric_name}"
+
+            metric_value = metrics.get(best_metric_name)
+
+            operator = np.greater if args.greater_is_better else np.less
+
+            self._should_upload_checkpoint = state.best_metric is None or operator(metric_value, state.best_metric)
```

Comment:
We possibly can override `self._should_upload_checkpoint` when we pass `log_checkpoints=True` and logging checkpoint configuration, can't we?

Reply:
Actually, there is nothing like `log_checkpoints=True`. Possible values are: `"same"`, `"last"`, `"best"` and `None`. So you have to choose a strategy -> on what event we should upload a checkpoint (assuming that it was saved locally by HuggingFace). `last` means that you would like to store only last checkpoint in Neptune. `same` that you would like to store every checkpoint saved by HuggingFace. `best` means that you would like to every checkpoint that produced better model and `None` just disables uploading model checkpoints.

Reply:
On the other hand `self._should_upload_checkpoint` only applies to `best` and tells that current checkpoint stores better model that was previously uploaded to Neptune.

Reply:
True